src/delfino/click_utils/pass_through_command.py

```diff
@@ -0,0 +1,109 @@
+import shlex
+from typing import Any, Callable, List
+
+import click
+
+from delfino.contexts import AppContext
+
+
+def pass_through_command(wrapped_command: str = None, **attrs: Any) -> Callable[[Callable], click.core.Command]:
+    """Creates a new command which passes arguments to the wrapped command.
+
+    Commands created by this decorator have `pass_through_args` parameter
+    which should be passed to the wrapped command as-is.
+
+    Example:
+    ```
+    @pass_through_command("pytest")
+    def typecheck(pass_through_args: List[str]):
+        run(["pytest", *pass_through_args])
+    ```
+
+    Note, this decorator will not automatically pass the argument to the wrapped
+    command. You need to do it manually.
+
+    The `pass_through_args` parameter can be specified in two ways:
+
+    (1) By `pass_through_args` config in the `pyproject.toml` file.
+
+    This command reads `pass_through_args` config from the config file and set
+    the configured value to the `pass_through_args` parameter.
+
+    The config must be placed under; `tool.delfino.plugins.COMMAND_NAME`.
+
+    Example:
+    ```
+    $ cat pyproject.toml
+    ...
+    [tool.delfino.plugins.test-unit]
+    pass_through_args = '--capture=no'
+    ...
+    $ delfino test-unit
+    -> run `pytest --capture=no`
+    ```
+
+    (2) By command line arguments
+
+    Example:
+    ```
+    $ delfino test-unit path_to_file
+    -> run `pytest path_to_file`
+
+    $ delfino test-unit --capture=no
+    -> raise error as `--capture` option is not defined
+
+    $ delfino test-unit -- --capture=no # All values after `--` will be considered as arg
+    -> run `pytest --capture=no`
+    ```
+
+    Args:
+        wrapped_command: The name of the command that arguments are passed to. Generates better help message when set.
+        attrs: Attributes to pass to click.command().
+
+    """
+
+    _argument_name = "pass_through_args"
```

Comment:
Constants are upper-case. This can also be placed in the top of the module as it is already marked as internal and doesn't need to be created every time this decorator is called.

```suggestion
    _ARGUMENT_NAME = "pass_through_args"
```